File: pset2/readability/readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<math.h>
int main(void)
{
    int letters = 0, words = 1, sentence = 0, i;
    string str = get_string("Enter the paragraph : ");

    for (i = 0; i < strlen(str); i++)
    {
        if ((str[i] >= 'a' && str[i] <= 'z') || (str[i] >= 'A' && str[i] <= 'Z'))
        {
            letters++;
        }
        else if (str[i] == ' ' || str[i] == '\t')
        {
            words++;
        }
        else if (str[i] == '.' || str[i] == '!' || str[i] == '?')
        {
            sentence++;
        }

    }

    float grade = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentence / (float) words) - 15.8;
    if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: pset6/readability/readability.py
from cs50 import get_string


def grade(para):
    letters = 0
    for c in para:
        if c.isalpha():
            letters += 1

    words_c = len(para.split())

    sentence = 0
    for stop in para:
        if stop == '.' or stop == '?' or stop == '!':
            sentence += 1

    grade = 0.0588 * (100 * letters / words_c) - 0.296 * (100 * sentence/words_c)-15.8
    if grade < 16 and grade >= 0:
        print(f"Grade : {round(grade)}")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


para = get_string("Enter the para : ")

grade(para)